src/components/ActivityFeed/ActivityFeed.js

```diff
@@ -87,46 +131,17 @@ Review.propTypes = {
   rating: number.isRequired,
 };
 
-// Check if a transition is the kind that
-// should be rendered in he ActivityFeed
-const shouldRenderTransition = transition => {
-  return [
-    TRANSITION_ACCEPT,
-    TRANSITION_CANCEL,
-    TRANSITION_COMPLETE,
-    TRANSITION_DECLINE,
-    TRANSITION_EXPIRE,
-    TRANSITION_REQUEST,
-    TRANSITION_REQUEST_AFTER_ENQUIRY,
-    TRANSITION_REVIEW_1_BY_CUSTOMER,
-    TRANSITION_REVIEW_1_BY_PROVIDER,
-    TRANSITION_REVIEW_2_BY_CUSTOMER,
-    TRANSITION_REVIEW_2_BY_PROVIDER,
-  ].includes(transition);
-};
-
-// Check if a user giving a review is related to
-// given tx transition.
-const isReviewTransition = transition => {
-  return [
-    TRANSITION_REVIEW_1_BY_CUSTOMER,
-    TRANSITION_REVIEW_1_BY_PROVIDER,
-    TRANSITION_REVIEW_2_BY_CUSTOMER,
-    TRANSITION_REVIEW_2_BY_PROVIDER,
-  ].includes(transition);
-};
-
-const hasUserLeftAReviewFirst = (userRole, lastTransition) => {
+const hasUserLeftAReviewFirst = (userRole, transaction, lastTransition) => {
+  const isProvider = txRoleIsProvider(userRole);
   return (
-    (lastTransition === TRANSITION_REVIEW_1_BY_CUSTOMER &&
-      userRole === TX_TRANSITION_ACTOR_CUSTOMER) ||
-    (lastTransition === TRANSITION_REVIEW_1_BY_PROVIDER &&
-      userRole === TX_TRANSITION_ACTOR_PROVIDER) ||
+    (isProvider && txHasFirstReviewBy(transaction, isProvider)) ||
+    (!isProvider && txHasFirstReviewBy(transaction, !isProvider)) ||
     areReviewsCompleted(lastTransition)
   );
 };
 
 const resolveTransitionMessage = (
+  transaction,
   transition,
   lastTransition,
```

Comment:
Remove this one. It is not needed since transaction is passed in.

Reply:
Also remove line 156 (`deliveredState`) and, in line 193, change `deliveredState` to `txIsCompleted(transaction)`
(Although, in my subbranch it will be renamed as `txIsDelivered`.)